### PR TITLE
Separate session context from session

### PR DESCRIPTION
**Pull request by einarmo**
This is a bit tricky, but it might allow the extractor to keep working even when connection to the server is lost. It should generally be fine to continue going with the _previous_ session context. If the setup is such that this won't work, the extractor should be configured to restart hard on connection loss. Most servers will work fine with this.

### CODE REVIEW

File: Extractor/UAClient.cs [hunk 1]
```diff
@@ -234,18 +231,9 @@ private async Task StartSession(int timeout = -1)
 
             // A restarted Session might mean a restarted server, so all server-relevant data must be cleared.
             // This includes any stored NodeId, which may refer to an outdated namespaceIndex
```
Comment (ozangoktan) [inline Extractor/UAClient.cs:233]: This comment does not make too much sense with the LoadAppConfig call

**Review by ozangoktan — COMMENTED** [review]
Generally looks good! Had a couple of comments.

File: Extractor/Nodes/UAVariable.cs [hunk 2]
```diff
@@ -532,9 +532,11 @@ public UAVariableMember(UAVariable parent, int index)
             TSParent = parent;
         }
 
-        public override string? GetUniqueId(IUAClientAccess client)
+        public override string? GetUniqueId(SessionContext? context)
         {
-            return client.GetUniqueId(Id, Index);
+            if (context == null) throw new InvalidOperationException("Attempt to get unique ID without initialized client");
```
[the remark this answers is not in the document]
Comment (einarmo) [reply Extractor/Nodes/UAVariable.cs:537]: Missed this one, context should never be nullable anymore.

File: Extractor/UAClient.cs (hunk 1 above)
Comment (einarmo) [reply Extractor/UAClient.cs:233]: No... I'll remove it.

**Review by ozangoktan — APPROVED** [review]
LGTM!

**Pull request merged**